core/dbt/adapters/base/impl.py

```diff
@@ -337,11 +337,12 @@ def _get_catalog_schemas(self, manifest: Manifest) -> SchemaSearchMap:
         # databases
         return info_schema_name_map
 
-    def _relations_cache_for_schemas(self, manifest: Manifest) -> None:
+    def _relations_cache_for_schemas(self, manifest: Manifest, cache_schemas: Set[BaseRelation] = None) -> None:
```

Comment:
Since we are here, let's just remove this function and move all of the logic into `set_relations_cache`

Reply:
Because these methods are defined in the `adapters` module, these are potentially breaking changes to the adapter interface. E.g. the Postgres plugin reimplements this method (currently with two arguments), and would require updating:
https://github.com/dbt-labs/dbt-core/blob/5e0a7659174a9277e2013390f8615acc97464d40/plugins/postgres/dbt/adapters/postgres/impl.py#L129-L131